BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m

```diff
@@ -504,45 +507,86 @@ - (void)connection:(NSURLConnection *)connection willSendRequestForAuthenticatio
             // Too many auth challenges, fail out
             BOXLog(@"Too many (%ld) authentication challenges - aborting.", (long)self.authChallengeCycles);
+            if (completionHandler) {
+                completionHandler(NSURLSessionAuthChallengeCancelAuthenticationChallenge, NULL);
+            } else {
+                [[challenge sender] cancelAuthenticationChallenge:challenge];
+            }
             NSString *message = NSLocalizedString(@"Unable to log in. Too many authentication challenges issued by the server.", @"Alert view message: message for failed Single-Sign-On login due to encountering too many authentication challenges (a technical network/server issue).");
             NSError *myError = [NSError errorWithDomain:NSURLErrorDomain code:NSURLErrorUserAuthenticationRequired userInfo:nil];
-            [self failAuthenticationWithConnection:connection challenge:challenge message:message error:myError];
+            [self failAuthenticationWithURLSessionTask:task message:message error:myError];
         } else {
             self.authChallengeCycles++;
 
             //  The NTLM protocol issues a 401 as part of the negotiation process.
-            //  In iOS 8, this 401 results in NSURLAuthenticationChallenge returning a "failureResponse".
+            //  In iOS 8 and later, this 401 results in NSURLAuthenticationChallenge returning a "failureResponse".
             //  However, NSURLAuthenticationChallenge does not return this in iOS 7.
             //  As seen from the CFNetworking logs, NSURLAuthenticationChallenge handles the 401 negotiation,
             //  and the process is seamless in iOS 7.
-            //  However, in iOS 8, the last step of the NTLM protocol returns an error that
+            //  However, in iOS 8 and later, the last step of the NTLM protocol returns an error that
             //  the developer must handle (as seen from CFNetworking and our app logs).
+            //  We only support versions of iOS newer than iOS 8.
             BOOL shouldApplyiOS8_NTLM_WAR = (self.isNTLMAuth &&
                                              [challenge previousFailureCount] > 0 &&
-                                             SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(@"8.0") &&
                                              (self.ntlmAuthFailures < kMaxNTLMAuthFailuresPriorToExit));
 
             // Handle the authentication challenge as a basic HTTP authentication challenge
             // (certificate-based, among other methods, is not currently supported).
             if (shouldApplyiOS8_NTLM_WAR) {
                 BOXLog(@"Applying iOS8 NTLM WAR, ntlmAuthFaulures = %ld", (long)self.ntlmAuthFailures);
                 self.ntlmAuthFailures++;
-                [[challenge sender] continueWithoutCredentialForAuthenticationChallenge:challenge];
+                if (completionHandler) {
+                    completionHandler(NSURLSessionAuthChallengeUseCredential, NULL);
```

Comment:
should this be NSURLSessionAuthChallengePerformDefaultHandling which will ignore the provided credential parameter automatically? is there an NTLM test env we can test with?